### ISSUE TITLE
Test of Package does not test all the classes in the package

### ISSUE THREAD

**Issue by sivasan8** [cbeust/testng#1940]
### TestNG Version
6.14.2

### Expected behavior
I have been constructing the XML via coding and i have added a test using a Package that has 7 classes and each class with @Test annotation. All the 7 test's should have been tested as per the design.

### Actual behavior
There we only 5 test/classes alone tested and the rest were simply ignored without any traces. 

### Is the issue reproductible on runner?

- [x] Eclipse
Yes its able to reproduce the issue on eclipse.

### Test Xml sample:

Please find attached xml file that has been found from your index.html. It clearly shows that all the classes inside the package are included but not all of them we tested.

[TestNG_Issue.txt](https://github.com/cbeust/testng/files/2524850/TestNG_Issue.txt)

**Comment by krmahadevan** [comment]
@sivasan8 

I took a look at the attached text file and this is all that it has

```xml
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE suite SYSTEM "http://testng.org/testng-1.0.dtd">
<suite name="BDR BVT">
  <test thread-count="5" name="Land At AUT Home">
    <packages>
      <package name="com.test.bvt.base"/>
      <package name="com.test.bvt.modules.home"/>
    </packages>
    <classes>
      <class name="com.test.bvt.base.TestActions"/>
      <class name="com.test.bvt.base.TestBase"/>
      <class name="com.test.bvt.base.TestRetry"/>
      <class name="com.test.bvt.base.TestUtilFunctions"/>
      <class name="com.test.bvt.modules.home.HomePage"/>
      <class name="com.test.bvt.modules.home.ProductInstallation"/>
    </classes>
  </test> <!-- Land At AUT Home -->
  <test thread-count="20" name="Temp_12">
    <packages>
      <package name="com.test.bvt.modules.vmbr"/>
    </packages>
    <classes>
      <class name="com.test.bvt.modules.vmbr.ClientBackupReport"/>
      <class name="com.test.bvt.modules.vmbr.ConfigureVMwareBackup"/>
      <class name="com.test.bvt.modules.vmbr.LiveVMRecovery"/>
      <class name="com.test.bvt.modules.vmbr.LiveVMRecoveryReport"/>
      <class name="com.test.bvt.modules.vmbr.MountBackup"/>
      <class name="com.test.bvt.modules.vmbr.RunIncremental"/>
      <class name="com.test.bvt.modules.vmbr.UnMountBackup"/>
    </classes>
  </test> <!-- Temp_12 -->
</suite> <!-- BDR BVT -->
```

With this information its not possible to troubleshoot what could be wrong. So request you to do the following and update

1. Please retry your scenario using TestNG's latest released version viz `7.0.0-beta1` and see if the problem persists.
2. If the problem persists, then please update this issue with a simple standalone project that can be used to recreate the problem (You would need to include the actual test classes, but while doing so, please help keep it simple and not have it use any other 3rd party libraries such as selenium etc.,)

**Comment by krmahadevan** [comment]
No answer from the user. Closing this ticket. If there's an update on this, please comment.
